### ISSUE TITLE
Projectors don't like refinement defined with previous field

### ISSUE THREAD

**Issue by kyoDralliam** [FStarLang/FStar#1276]
Found by @kkohbrok.

```F#
module Test

assume val type_mapper: n:nat -> Type0

(* Without this option the inductive won't go through *)
#set-options "--__temp_no_proj Test"

abstract noeq type container =
  | TC:
    pred:(n:nat -> t:Type0) ->
    n:(let t i = pred i in k:nat{type_mapper k == i:nat{t i}}) ->
    container

(* works *)
let my_proj__TC__item__n = fun projectee -> let TC _ n = projectee in n

(* works *)
let __proj__TC__item__pred projectee = let TC pred _ = projectee in pred
(* fail *)
let proj__TC__item__n : c:container -> (k:nat{type_mapper k == i:nat{__proj__TC__item__pred c i}})= fun projectee -> let TC _ n = projectee in n
```

The error is 
```
Subtyping check failed; expected type (k:Prims.nat{ Test.type_mapper k == (i:Prims.nat{ TC?.pred projectee i }) }); got type (k:Prims.nat{ Test.type_mapper k == (i:Prims.nat{ pred i }) })
```
The problem seem to stem from the fact that we cannot prove *equality* of two refined types `x:t{phi}` and `x:t{phi'}` even if `phi` and `phi'` are definitionally equal (in the case at hand the first reduces to the second).

**Comment by kkohbrok** [comment]
Recently, I had another problem that might be related:
```
module Test

noeq type type_container =
  | TC:
    t: Type0 ->
    type_container

#set-options "--z3rlimit 1000 --max_ifuel 4 --max_fuel 4"
let create() =
  let tc = TC int in
  let t1 = tc.t in
  let t2 = int in
  assert(t1 == t2);
  assert(i1:(t1){True} == i2:(t2){True})
```
The remedy was the following lemma. A little clunky, but it helped.
```
val lemma_type_equality: t1:Type0 -> t2:Type0 -> pred:(t1 -> Type0) -> Lemma
  (requires t1 == t2)
  (ensures (t1 == t2 ==> i:t1{pred i} == i:t2{pred i}))
let lemma_type_equality t1 t2 pred = ()
```

**Comment by aseemr** [comment]
This is a projectors issue that is fixed as part of #1355. I have added the testcases there, closing this issue.
